### DISCUSSION TITLE
Depth bug when importing YML

### DISCUSSION THREAD

**Discussion by kaelsonofkrypto** [nagadomi/nunif#169] [General]
i finally was able to test importing my own depth maps in yml, i deflickered the iw3 depth maps in davinci and then i replaced the files in the  export disparity "depth" folder with my new deflickered maps, and imported the yml file in iw3. So what i found out is depth 3.0 was the max u can push before the image begins aliasing/jagged edges. i also noticed the depth looked no different on 5.0 depth but significatly looked worse with  jagged edges, ironically became lower resolution, and got flatter in background when going above 3.0 on depth, so i had a hunch, i lowered depth to 2.5 and image became suddenly more 3d-ish and clearer, then when i lowered to 1.8 depth, image became even more 3d-ish. so the depth function doesnt not work properly when using YML method. So on yml method its better to use a low depth of 1.8. essentially 1.8 depth on yml method looks equivalent to 3.0 depth on the regular 2d to 3d method.  The regular 2d to 3d function doesnt have this bug , it works fine if u just do a regular 2d to 3d conversion with no yml, but the moment u use yml method u actually have to use less depth for it to function properly

**Comment by diverswan9** [comment]
I've been experimenting with .yml as well. Just to see, I loaded a bunch of different models and they all came out pretty much the same. It must be mainly using the depthmaps for what it needs.

**Comment by nagadomi** [comment]
If there is no problem importing without editing after export, then it is a problem of editing.

One cause that I can think of is the resolution (image size) of depth images.
The row_flow_v3/v2 model is performed on the depth input resolution. It produces a larger stretch the larger the image.
If you have resized the depth image for editing, you will need to resize it back to the original size to get the same result.

**Reply by kaelsonofkrypto** [reply]
the source file i use a 4k image, when you do an export disparity in iw3 and u use a 4k source the rgb files will export to 3840x2160p but on the depth map it will export to the resolution of 518 when u use depth resolution 518 which is what gives a proper depth map since thats the max that any models were trained on. If u try to import those files back via yml, its not going to work because iw3 will not work on  a depth map being a different resolution than the rgb even tho thats what it exports to, so i did resize the depth map to match to rgb file size. so thats not the issue because if i left it the 518 it wouldnt work back on import,

**Reply by nagadomi** [reply]
iw3 works with different resolutions for rgb and depth. Because in most cases they are different resolutions.
I think you are confusing it with 3DCombine or some other software.

**Reply by kaelsonofkrypto** [reply]
yea im familiar with the limitations of combine,  a few other people i chat with in a 3d community, did export disparity and posted screenshots showing if they had a different resolution on depth map vs the rgb files it would post an error, so to avoid this error u have to upscale it to the same size. but this is not the issue, im just pointing out that the bug is on the depth, i already figured out u have to lower depth on yml to work within this bug, , the setting is working in reverse on yml, its not doing the same as the regular 2d to 3d if u let iw3 handle everything. not many people have tested this function because there wasnt any clear instructions on how to do the yml method until someone else in our 3d community showed replacing the files

**Reply by nagadomi** [reply]
In addition, forward_fill is performed on rgb(source) resolution. So it should not be affected by resizing the depth image.
(forward_fill has a problem that some areas of the output will be pixelized, though.)

**Reply by kaelsonofkrypto** [reply]
yes this part i did tell others in the community that forward fill will cause pixelation because i remember u saying something regarding the resolution of it which is why i started only using row flow 3

**Reply by nagadomi** [reply]
>so does row flow 3 and 2 not work properly on export disparity?

row_flow_v3 probably works fine up to depth image width=1920 (but results differ depending on image size of depth).
That is the maximum resolution used when training row_flow_v3 model.

**Reply by kaelsonofkrypto** [reply]
oh my God Really lol , so then if i use a source higher than that 1920 max u mentioned, could this be leading to artifacts since 1920 is max that row flow flow was trained on? oh i just noticed you said depth image, so does this apply only to the depth map and not the source? sorry for the question, i just want to also make sure i relay the information correctly to others as well, and i guarantee others in the community im in don't know this. would you be able to document this in the read me on the max resolutions. 

Some other things im confused on with yml, i want to make sure i understand this right, when generating depth map u should leave it vanilla meaning have zero on edge fix and forescale and converge, only thing u will set is the depth and when u import the yml later after cleaning depth map, u will put your values of the edge fix and foreground?  i tried to look up information in the readme file about yml and how it works but i couldnt find anything on it in the readme

**Reply by nagadomi** [reply]
row_flow_v3/v2 is performed on depth images. width=1920 limit is only for depth image sizes.
For edge fix and foreground scale, see https://github.com/nagadomi/nunif/discussions/160

**Reply by kaelsonofkrypto** [reply]
thanks for the feed, this is good information that would be beneficial i believe in the readme section where u have info on different sections of the program.

There are a few questions i had that i couldn't find info on that feed. 

1.For the depth resolution setting, depth level, and the converge, do those 3 get applied on export disparity to depth map or do they get applied on import of yml? 

2. i know u mentioned that row flow is trained on 1920, so then if i use a 4k or 8k source, then should i be selecting an output limit of 1080p? is this why there isn't an output limit of 4k option available in that output limit section?

3.  also does tta get applied during export disparity or only on import, and does it make more sense to use fp16 on export disparity and tta on import yml? trying to ensure the depth map and conversion is accurate as possible but if its not working on a certain phase i want to make sure i apply tta to the correct phase. i do like tta as i can tell the different, it has less flickering than fp16

4. also on the feed i read you mentioned edge fix (edge dilation) and mapper (foreground) get applied on depth which i assume you mean this gets applied to the depth map when you do export disparity, and it does not get applied on video conversion which i assume is basically the process of you importing the yml file, and also flicker reduction (Ema normalize) gets applied on both export disparity of depth map and on import of yml for video generation. Did i understand this correctly?

and please forgive me for all these questions, im just trying to make sure i get this right.

**Reply by nagadomi** [reply]
>2

I added the `Stereo Processing Width` (`--stereo-width`) option.

This allows you to specify the size (width) of the depth image to be input to `row_flow_v3`/v2. Internally, the depth image is resized to the specified size before processing.
For example, if importing 2K depth images, you can specify 1280 or 1920 to be processed as a safe size. By specifying this option, even if the depth image is 2k to be imported, `2` is not needed.
Also, it can be used when you want to run `row_flow_v3` at a higher resolution than `Depth Resolution`. (Not tested in detail, but may be able to reduce artifacts)

>1

`Depth Resolution` is used only when `Export`, and `3D Strength` and `Convergence Plane` are only used when Import (yml input).

>3

`TTA` is only used when `Export`.

Basically, `Export/Export disparity` only outputs depth, so the stereo generation related settings are not used.
The confusing parts are `Foreground Scale` and `Edge Fix`. They are applied at different times in `Export` and `Export disparity`. However, they are only applied when either Exporting or Importing.

**Reply by kaelsonofkrypto** [reply]
Thanks so much, appreciate the addition and the explanation. Can't wait to try it  out.

**Reply by diverswan9** [reply]
Is that update supposed to be  for export/export disparity only? Can it be used for a normal 2D-3D conversion? I tried applying different settings for 'Stereo Processing Width' on a normal conversion (not an export) and didn't work so well. I

**Reply by nagadomi** [reply]
The reason for adding `Stereo Processing Width` is for Import (yml input, solution for cases where the depth image is too large).
It can be used for other than Import, but the effect is different in the case of downscaling and upscaling.
For non-Import uses, it is (bilinear) upscaling in most cases, it may not make much difference, or it may be worse.
If you do not feel it makes sense to use it, I recommend `Default`.
`Default` means do nothing. It is the same behavior as before.
